staging/src/k8s.io/client-go/util/workqueue/rate_limiting_queue.go

```diff
@@ -41,9 +45,16 @@ func NewRateLimitingQueue(rateLimiter RateLimiter) RateLimitingInterface {
 	}
 }
 
+// NewNamedRateLimitingQueue constructs a new workqueue with given name.
 func NewNamedRateLimitingQueue(rateLimiter RateLimiter, name string) RateLimitingInterface {
+	return NewNamedRateLimitingQueueWithCustomClock(rateLimiter, name, clock.RealClock{})
```

Comment:
If we would put your queue directly to workqueue package, you wouldn't need those changes.

Reply:
I agree that we want to ultimately move this struct to workqueue, but I think that we should first resolve the naming conflict with "RateLimitingQueue" available there.

Personally, I would rename methods in that struct:
* AddRateLimited -> AddRetryAttempt or similar
* Forget -> Succeeded/FinishedRetries() or similar

This way, If I add another RateLimitingQueue to that place it would be clear what the difference is.

This is clearly quite huge refactoring (there is a lot of users of this class everywhere) so I don't think we can do that for now.

Alternatively, I can add a huge comment to existing AddRateLimited function and my rate limiting functions to describe why "rate limiting" in this method means something different than in that method, but I think it will be error prone.